cpp/util/diagonalize.cc

```diff
@@ -0,0 +1,27 @@
+#include "diagonalize.h"
+
+namespace freud { namespace util {
+
+void diagonalize33SymmetricMatrix(float mat[9], float eigen_vals[3], float eigen_vecs[9])
+{
+    Eigen::Matrix3f m = Eigen::Map<Eigen::Matrix3f> (mat);
+ 
+    Eigen::SelfAdjointEigenSolver<Eigen::Matrix3f> es;
+    es.compute(m);
+
+    if (es.info() != Eigen::Success)
+    {
+        // numerical issue, return identity matrix
+        Eigen::Matrix3f id = Eigen::Matrix3f::Identity();
+        Eigen::Map<Eigen::Matrix3f> (eigen_vecs, 3, 3) = id;
+        // set eigenvalues to zero so it's easily detectable
+        eigen_vals[0] = eigen_vals[1] = eigen_vals[2] = 0.0;
+    }
+    else
+    {
+        Eigen::Map<Eigen::Matrix3f> (eigen_vecs, 3, 3) = es.eigenvectors();
```

Comment:
I don't think we need the `3, 3` or the `3` below, because these are fixed-size matrix types.

> A fixed-size read-only vector of integers might be declared as
> `Map<const Vector4i> mi(pi);`
> where `pi` is an `int *`. In this case the size does not have to be passed to the constructor, because it is already specified by the Matrix/Array type.

https://eigen.tuxfamily.org/dox/group__TutorialMapClass.html

Reply:
This didn't work - we'd have to cast the results of `es` to the fixed-size types, too. I just reverted my change and we'll merge as-is.